audio_service/lib/audio_service.dart

```diff
@@ -940,35 +946,83 @@ class AudioService {
     return handler;
   }
 
+  /// Allows to specify a URI for an album art (and optionally headers) that will be
+  /// used when [MediaItem.artUri] is null, or loading the art has failed.
+  ///
+  /// All schemes that are supported by [MediaItem.artUri] are supported here as well.
+  static Future<void> setFallbackArt(
+    Uri uri, {
+    Map<String, String>? headers,
+  }) async {
+    assert(_cacheManager != null, 'Call AudioService.init first');
+    _updateFallbackArtCache = true;
+    _fallbackArtUri = uri;
+    _fallbackArtHeaders = headers;
+    if (__handler != null) {
+      final currentMediaItem = __handler!.mediaItem.value;
+      if (currentMediaItem != null) {
+        await AudioService.cacheManager.removeFile(uri.toString());
+        await _updateMediaItem(currentMediaItem);
+      }
+    }
+  }
+
+  static bool _updateFallbackArtCache = false;
+  static Uri? _fallbackArtUri;
+  static Map<String, String>? _fallbackArtHeaders;
+  static Object? _artFetchOperationId;
```

Comment:
Any reason why you reversed my change to maintain this state in the closure rather than keep everything as a global variable?

Reply:
I did this to make `_updateMediaItem` a separate function.